File: realtime/internal/player.go
package realtime

type PlayerRole string

const (
	RoleHost   PlayerRole = "HOST"
	RolePlayer PlayerRole = "PLAYER"
)

type PlayerStatus string

const (
	StatusJoining      PlayerStatus = "JOINING"
	StatusConnected    PlayerStatus = "CONNECTED"
	StatusDisconnected PlayerStatus = "DISCONNECTED"
)

type PlayerProfile struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

type PlayerInfo struct {
	Role    PlayerRole    `json:"role"`
	Profile PlayerProfile `json:"profile"`
	Status  PlayerStatus `json:"status"`
}

type Player interface {
	ChangeStatus(status PlayerStatus)
	Role() PlayerRole
	ChangeRole(r PlayerRole)
	ID() string
	Info() *PlayerInfo
}

File: frontend/types/player.ts
import { GameRole } from "./game";

export type Player = {
	id: string;
	name: string;
	role: PlayerRole;
	status: PlayerConnectionStatus;
	avatarSeed: string;
	avatarColor: string;
	score: number;
	gameRole: GameRole;
};

export enum PlayerRole {
	HOST = "HOST",
	PLAYER = "PLAYER",
}

export enum PlayerConnectionStatus {
	JOINING = "JOINING",
	CONNECTED = "CONNECTED",
}


